aiidalab_widgets_base/process.py

```diff
@@ -500,38 +481,66 @@ def update(self, _=None):
-                                      process_label=None,
+                                      process_state=self.process_states,
+                                      process_label=self.process_label,
                                       exit_status=None,
                                       failed=None)
         relationships = {}
-        if self.incoming_node.value:
-            relationships = {**relationships, **{'with_outgoing': load_node(int(self.incoming_node.value))}}
+        if self.incoming_node:
+            relationships = {**relationships, **{'with_outgoing': self.incoming_node}}
 
-        if self.outgoing_node.value:
-            relationships = {**relationships, **{'with_incoming': load_node(int(self.outgoing_node.value))}}
+        if self.outgoing_node:
+            relationships = {**relationships, **{'with_incoming': self.outgoing_node}}
 
         query_set = builder.get_query_set(
             filters=filters,
-            past_days=None if self.past_days.disabled else self.past_days.value,
+            past_days=None if self.past_days < 0 else self.past_days,
             order_by={'ctime': 'desc'},
             relationships=relationships,
         )
-        projected = builder.get_projected(query_set,
-                                          projections=['pk', 'ctime', 'process_label', 'state', 'process_status'])
+        projected = builder.get_projected(
+            query_set, projections=['pk', 'ctime', 'process_label', 'state', 'process_status', 'description'])
         dataf = pd.DataFrame(projected[1:], columns=projected[0])
         self.output.value = "{} processes shown".format(len(dataf))
+
+        # Add HTML links.
         dataf['PK'] = dataf['PK'].apply(
             lambda x: """<a href={0}aiidalab-widgets-base/process.ipynb?id={1} target="_blank">{1}</a>""".format(
                 self.path_to_root, x))
         self.table.value += dataf.to_html(classes='df', escape=False, index=False)
 
+    @validate('incoming_node')
+    def _validate_incoming_node(self, provided):
+        """Validate incoming node. The function load_node takes care of managin ids and uuids."""
+        if provided['value']:
+            try:
+                return load_node(provided['value'])
+            except (MultipleObjectsError, NotExistent):
+                return None
+        return None
+
+    @validate('outgoing_node')
+    def _validate_outgoing_node(self, provided):
+        """Validate outgoing node. The function load_node takes care of managin ids and uuids."""
+        if provided['value']:
+            try:
+                return load_node(provided['value'])
+            except (MultipleObjectsError, NotExistent):
+                return None
+        return None
+
+    @validate('process_label')
+    def _validate_process_label(self, provided):
+        if provided['value']:
+            return provided['value']
+        return None
+
     def _follow(self, update_interval=10):
```

Comment:
```suggestion
    def _follow(self, update_interval):
```
The calling function has the default value, otherwise this is slightly overspecified.